### PR TITLE
Extend NcInputField props and forward $attrs to NcPasswordField and NcTextField

### PR DESCRIPTION
**Pull request by skjnldsv**
https://deploy-preview-3485--nextcloud-vue-components.netlify.app/#/Components/NcFields

Fix https://github.com/nextcloud/nextcloud-vue/issues/3479

### CODE REVIEW

File: src/components/NcInputField/NcInputField.vue [hunk 1]
```diff
@@ -19,6 +19,19 @@
   - along with this program. If not, see <http://www.gnu.org/licenses/>.
 -->
 
+<docs>
+### Description
+
+This component is used by the other Fields components.
+It extend and style an HTMLInputElement.
```
Comment (nickvergessen) [inline src/components/NcInputField/NcInputField.vue:26]:
```suggestion
It extends and styles an HTMLInputElement.
```

File: src/components/NcInputField/NcInputField.vue [hunk 2]
```diff
@@ -19,6 +19,19 @@
...
+
+You cannot use it as is. This is here for documentation purposes.
+See the other fields components.
```
Comment (nickvergessen) [inline src/components/NcInputField/NcInputField.vue:29]:
```suggestion
See the other field components.
```

**Review by nickvergessen — APPROVED** [review]
:electron:

**Review by ShGKme — COMMENTED** [review]
It seems to be a breaking change around `minlength` in `NcPasswordField`

File: src/components/NcPasswordField/NcPasswordField.vue [hunk 3]
```diff
@@ -337,12 +194,14 @@ export default {
 			}
 			return this.internalHelpMessage
 		},
+
 		rules() {
 			const { minlength, passwordPolicy } = this
 			return {
 				minlength: minlength ?? passwordPolicy?.minLength,
 			}
 		},
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/NcPasswordField/NcPasswordField.vue:221]: Nice catch! Thanks @ShGKme :)

**Review by ShGKme — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

File: src/components/NcInputField/NcInputField.vue [hunk 4]
```diff
@@ -116,11 +129,20 @@ export default {
 		},
 
 		/**
-		 * The input element type
+		 * The type of the input element, it can be `text`, `password`,
+		 * `email`, `tel` and `url`.
```
Comment (skjnldsv) [inline src/components/NcInputField/NcInputField.vue:133]:
```suggestion
		 * The type of the input element.
```

**Pull request merged**